### PR TITLE
Visual tests actions

### PR DESCRIPTION
**Pull request by akalburgi**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs [hunk 1]
```diff
@@ -842,7 +842,7 @@ scenarioTests mbScenarioService = Tasty.testGroup "Scenario tests"
 
 visualDamlTests :: Tasty.TestTree
 visualDamlTests = Tasty.testGroup "Visual Tests"
-    [   testCase' "Set files of interest" $ do
+    [   testCase' "Templete with no actions (edges) from choices" $ do
```
Comment (rohanjr) [inline compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs]:
```suggestion
    [   testCase' "Template with no actions (edges) from choices" $ do
```

File: compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs [hunk 2]
```diff
@@ -864,20 +870,87 @@ visualDamlTests = Tasty.testGroup "Visual Tests"
...
                 , "    signatory owner"
                 , "    controller owner can"
-                , "      nonconsuming ReducedCoin : ContractId Coin"
+                , "      nonconsuming ReducedCoin : ()"
                 , "        with otherCoin : ContractId Coin"
                 , "        do "
-                , "        cn <- fetch otherCoin "
-                , "        create this with amount = cn.amount - 10"
+                , "        cn <- fetch otherCoin"
+                , "        archive otherCoin"
```
Comment (hurryabit) [inline compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs]: Do we need the `archive` here if we just want to ensure that `fetch` does not produce an edge in the graph?

File: compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs [hunk 3]
```diff
@@ -864,20 +870,87 @@ visualDamlTests = Tasty.testGroup "Visual Tests"
...
             expectNoErrors
             expectedTemplatePoperties fetchTest $ Set.fromList
-                [TemplateProp (Set.fromList
+                [TemplateProp "Coin"
+                    (Set.fromList
                     [   ExpectedChoices "Archive" True,
                         ExpectedChoices "ReducedCoin" False
                     ])
-                    1
+                    (Set.fromList [Exercise "F:Coin" "Archive"])
```
Comment (hurryabit) [inline compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs]: The expectation should say which choice is expected to exercise the `Archive` choice of `F:Coin`.

File: compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs [hunk 4]
```diff
@@ -118,13 +119,22 @@ data ShakeTestEnv = ShakeTestEnv
...
     { _cName :: String
     , _consuming :: Bool
     } deriving (Eq, Ord, Show )
+
 data TemplateProp = TemplateProp
-    { _choices :: Set.Set ExpectedChoices
-    , _action :: Int
+    { _tplName :: T.Text
+    , _choices :: Set.Set ExpectedChoices
+    , _action :: Set.Set ExpectedChoiceAction
```
Comment (hurryabit) [inline compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs]: The `_action` field should be moved into `ExpectedChoices`.

File: compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs [hunk 5]
```diff
@@ -118,13 +119,22 @@ data ShakeTestEnv = ShakeTestEnv
...
     }
 
+type TemplateName = String
+type ChoiceName = String
+
+data ExpectedChoiceAction
+    = Create TemplateName
+    | Exercise TemplateName ChoiceName deriving (Eq, Ord, Show)
+
 data ExpectedChoices = ExpectedChoices
```
Comment (hurryabit) [inline compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs]:
```suggestion
data ExpectedChoice = ExpectedChoice
```
This is just about one choice, right?

File: compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs (hunk 4 above)
Comment (akalburgi) [reply compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs]: Yup, thanks will move it

File: compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs (hunk 2 above)
Comment (akalburgi) [reply compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs]: Nope,  got rid of it

File: compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs (hunk 3 above)
Comment (akalburgi) [reply compiler/damlc/tests/src/DA/Test/ShakeIdeClient.hs]: Got rid of that action we this test intent was to make sure fetch did not add a edge

**Pull request merged**